ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs

```diff
@@ -245,38 +249,50 @@ decreaseLimit (Limit 0) = Nothing
 decreaseLimit (Limit n) = Just . Limit $ n - 1
 
 processAll ::
-     forall blk b st. HasHeader blk
+     forall blk b st. (HasHeader blk, HasAnnTip blk)
   => Either (ImmutableDB IO blk) (ChainDB IO blk)
   -> ResourceRegistry IO
   -> BlockComponent blk b
+  -> ExtLedgerState blk
   -> Limit
   -> st
   -> (st -> b -> IO st)
   -> IO st
 processAll = either processAllImmutableDB processAllChainDB
 
 processAll_ ::
-     forall blk b. HasHeader blk
+     forall blk b. (HasHeader blk, HasAnnTip blk)
   => Either (ImmutableDB IO blk) (ChainDB IO blk)
   -> ResourceRegistry IO
   -> BlockComponent blk b
+  -> ExtLedgerState blk
   -> Limit
   -> (b -> IO ())
   -> IO ()
-processAll_ db rr blockComponent limit callback =
-    processAll db rr blockComponent limit () (const callback)
+processAll_ db registery blockComponent initLedger limit callback =
+    processAll db registery blockComponent initLedger limit () (const callback)
 
 processAllChainDB ::
-     forall st blk b. HasHeader blk
+     forall st blk b. (HasHeader blk, HasAnnTip blk)
   => ChainDB IO blk
   -> ResourceRegistry IO
   -> BlockComponent blk b
+  -> ExtLedgerState blk
   -> Limit
   -> st
   -> (st -> b -> IO st)
   -> IO st
-processAllChainDB chainDB rr blockComponent limit initState callback = do
-    itr <- ChainDB.streamAll chainDB rr blockComponent
+processAllChainDB chainDB registery blockComponent (ExtLedgerState {..}) limit initState callback = do
```

Comment:
Typo: `registery` should be `registry`

(This is elsewhere in the module too.)